src/feeder.ts

```diff
@@ -52,52 +52,46 @@ export default class Feeder {
     );
   }
   /**
-   * Fetch the RSS feed.
+   * Fetch the RSS.
    */
   async getRss() {
-    return fetch(this.url)
-      .then((response) => response.text())
-      .finally(() => {
-        console.log(
-          `${chalk.green('✔')} RSS feed fetched`
-        );
-      });
+    const response = await fetch(this.url);
+
+    if (!response.ok) {
+      throw new Error(
+        `Fetch RSS from ${chalk.cyan(this.url)} failed`
+      );
+    }
+
+    const data = await response.text();
+
+    return data;
   }
   /**
    * Check if the output directory exists and
    * create it if it doesn't.
    */
-  async checkOutDir() {
+  async createDir() {
     const dir = path.dirname(this.file);
-    const recursive = true;
+    const hasFile = await fs.existsSync(dir);
 
-    if (await fs.existsSync(dir)) {
+    if (hasFile) {
       return;
     }
 
-    await fs.mkdir(
+    await fs.mkdirSync(
```

Comment:
You youd use fs.promises.mkdir that returns a Promise so you'd use await instead